pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -39,7 +39,7 @@ type state interface {
 	CanTransitionTo(next state) bool
 	// Executes this state, returning a followup state to be immediately executed as well.
 	// The 'noOp' state should be returned if the state has no followup.
-	Execute(msg dispatcher.DIDCommMsg, ctx context) (followup state, err error)
+	Execute(msg dispatcher.DIDCommMsg, thid string, ctx context) (followup state, err error)
```

Comment:
@rolsonquadras @fqutishat @talwinder50 @llorllale I'm wondering if thid could make sense in DIDCommMsg?

(I see these arguments paired together in the PR).

Reply:
It would surely be a good candidate for ID inside DIDCommMsg. The one problem i see is that it is not consistent in different DIDComm messages (DIDCommMsg is constructed at inbound transport layer). `~thread` decorator is not present in exchangeRequest. The DIDExchange Service derives from `id` attribute. Apart from this, I believe all DIDComm message types will have `~thread` decorator.